### PR TITLE
Add `tracking` keyword support for `cutout.irradiation`

### PR DESCRIPTION
**Pull request by Tasqu**
## Change proposed in this Pull Request

Add `tracking` keywords to `cutout.irradiation` functions similar to `cutout.pv`.

## Description

The underlying `SurfaceOrientation` and `TiltedIrradiation` functions already have support for solar tracking options. However, there was no way of accessing this functionality through the `cutout.irradiation` method.

## Motivation and Context

Allows access to the irradiation quantities of surfaces with solar tracking. Admittedly, this is a pretty niche use case, but easy to implement and still useful for some applications.

## How Has This Been Tested?

Not very thoroughly. I used the `building_stock_weather_aggregation.ipynb` example to briefly test that the `tracking="vertical"` keyword in `cutout.irradiation` impacts the output irradiation quantities as expected. Other than that, I'm relying on the underlying `SurfaceOrientation` and `TiltedIrradiation` functions.

I tried running `pytest`, but it seems the current `master` has some issues, and the tests seem to get stuck at `test/test_preparation_and_conversion.py::TestERA5::test_data_module_arguments_era5`. I'm guessing this might be related to the Copernicus CDS undergoing maintenance?

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [x] I locally ran `pytest` inside the repository and no unexpected problems came up.
- [x] I have adjusted the docstrings in the code appropriately.
- [x] I have documented the effects of my code changes in the documentation `doc/`. *(No changes to the documentation needed as far as I can tell)*
- [x] I have added newly introduced dependencies to `environment.yaml` file. *(No new dependencies)*
- [x] I have added a note to release notes `doc/release_notes.rst`.
- [x] I have used `pre-commit run --all` to lint/format/check my contribution

### CODE REVIEW

**Comment by euronion** [comment]
Thanks @Tasqu 

@FabianHofmann do you have time to review?

**Review by fneum — APPROVED** [review]
This looks good to me.

File: atlite/convert.py [hunk 1]
```diff
@@ -539,23 +539,34 @@ def wind(cutout, turbine, smooth=False, add_cutout_windspeed=False, **params):
 
 # irradiation
 def convert_irradiation(
-    ds, orientation, irradiation="total", trigon_model="simple", clearsky_model=None
+    ds,
+    orientation,
+    tracking,
```
Comment (fneum) [inline atlite/convert.py:544]: Maybe `tracking=None` in order not to introduce a new compulsory argument?

Comment (Tasqu) [reply atlite/convert.py]: Good point. Done.

**Review by fneum — APPROVED** [review]

**Pull request merged**